genienlp/models/transformer_token_classification.py

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright (c) 2021 The Board of Trustees of the Leland Stanford Junior University
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+logging.getLogger("filelock").setLevel(logging.WARNING)
+
+from transformers import AutoModelForTokenClassification, AutoConfig
+
+from ..models.base import GenieModel
+from ..util import adjust_language_code
+from . import TransformerSeq2Seq
+from ..data_utils.numericalizer import TransformerNumericalizer
+
+logger = logging.getLogger(__name__)
+
+
+class TransformerForTokenClassification(TransformerSeq2Seq, GenieModel):
```

Comment:
`TransformerSeq2Seq` is already a subclass of `GenieModel`, no need for multiple inheritance.
Also, I think more of this class's code can be inherited from its parent and reused, instead of copying their code. Can you clean it up a bit?

Reply:
I need to initialize TransformerForTokenClassification from GenieModel not TransformerSeq2Seq cause it'll load unwanted things. I think `super(TransformerSeq2Seq, self).__init__(self, config)` is equivalent to `GenieModel.__init__(self, config)` but wasn't sure which one is better. Thought the second one may be more explicit. What is your suggestion?

Reply:
Why is `TransformerForTokenClassification` a subclass of `TransformerSeq2Seq` at all? In addition to the fact that it conceptually doesn't make sense (is-a relationship doesn't hold), you are not using any of the code of the parent in the child.
Isn't it better to just inherit from `GenieModel`?

Reply:
Hmm, yeah I think initially I was trying to reuse some methods from `TransformerSeq2Seq` like `add_new_vocab_from_data` and `confidence_features` but `add_new_vocab_from_data` is copied in and confidence is not supported yet for this model, so yeah it should be fine to use GenieModel.